server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -566,6 +567,12 @@ private CompletionStage<RestResponse> getCacheDistribution(RestRequest request)
             .thenApply(distributions -> asJsonResponse(Json.array(distributions.stream().map(CacheDistributionInfo::toJson).toArray())));
    }
 
+   private CompletionStage<RestResponse> getKeyDistribution(RestRequest request) {
+      return CompletableFuture.supplyAsync(() -> keyDistribution(request), invocationHelper.getExecutor())
```

Comment:
This shouldn't need supplyAsync as the `keyDistribution` is already non blocking right?

Reply:
Yes.